src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -241,11 +239,9 @@ def get_pyenv_root(env: Environment) -> str | None:
 
 
 @rule
-async def python_bootstrap(python_bootstrap_subsystem: PythonBootstrapSubsystem) -> PythonBootstrap:
-    env_tgt = await Get(
-        EnvironmentTarget,
-        EnvironmentRequest(LOCAL_ENVIRONMENT_MATCHER, description_of_origin="<infallible>"),
-    )
+async def python_bootstrap(
+    python_bootstrap_subsystem: PythonBootstrapSubsystem, env_tgt: EnvironmentTarget
+) -> PythonBootstrap:
```

Comment:
@chrisjrn this code will move into your subsystem creation code path. But for now, this is a big improvement :)